---
title: "3. Sequence database"
author: "Permentier"
date: "2024-06-14"
output: html_document
---

# Sequence database

## Libraries

```{r}
suppressPackageStartupMessages({
  library(dplyr)
  library(tidyr)
  library(Biostrings)
})
```

## FASTA files

This FASTA file contains the 54384 protein sequences that DIA-NN used for identifying + quantifying M. capitata proteins in the samples (http://cyanophora.rutgers.edu/montipora/):
```{r}
setwd("E:/PhD_YEAR4/Proteomics/3. Sequence_database")
FASTA_Mcap <- readAAStringSet("FASTA_files/Montipora_capitata_HIv3.genes.pep.FASTA")
head(names(FASTA_Mcap))
length(FASTA_Mcap)
```

This FASTA file contains the 35913 protein sequences that DIA-NN used for identifying + quantifying Cladocopium sp. proteins in the samples (http://cyanophora.rutgers.edu/montipora/):
```{r}
FASTA_symbiont <- readAAStringSet("FASTA_files/SymbC1.Gene_Models.PEP.FASTA")
head(names(FASTA_symbiont))
length(FASTA_symbiont)
```

This FASTA file contains the 125 protein sequences that DIA-NN used for identifying + quantifying cRAP proteins. cRAP means common Repository of Adventitious Proteins, these are common contaminant proteins such as e.g. human skin proteins, latex glove rubber proteins, sheep whool proteins, proteins used in research, etc. DIA-NN automatically used this FASTA file from the Cambridge Centre for Proteomics (https://proteomics.bio.cam.ac.uk/).
```{r}
FASTA_cRAP <- readAAStringSet("C:/DIA-NN/1.9/camprotR_240512_cRAP_20190401_full_tags.FASTA")
head(names(FASTA_cRAP))
length(FASTA_cRAP)
```

## Sequence database

This data frame contains all sequence information for this analysis.
```{r}
FASTA_df <- rbind(tibble(FASTA_header = names(FASTA_Mcap),Sequence = as.character(FASTA_Mcap)) %>%
                    mutate(Organism = "Montipora capitata"),
                  tibble(FASTA_header = names(FASTA_symbiont),Sequence = as.character(FASTA_symbiont)) %>%
                    mutate(Organism = "Cladocopium"),
                  tibble(FASTA_header = names(FASTA_cRAP),Sequence = as.character(FASTA_cRAP)) %>%
                    mutate(Organism = "cRAP"))
FASTA_df
```

Write all information to a csv file that we will use in subsequent steps.
```{r}
#write.csv(FASTA_df,file="sequence_database.csv",row.names = FALSE)
```




